exporter/awsxrayexporter/internal/translator/segment.go

```diff
@@ -60,16 +61,173 @@ const (
 	identifierOffset = 11 // offset of identifier within traceID
 )
 
+const (
+	localRoot = "LOCAL_ROOT"
+)
+
 var (
 	writers = newWriterPool(2048)
 )
 
+// MakeSegmentDocuments converts spans to json documents
+func MakeSegmentDocuments(span ptrace.Span, resource pcommon.Resource, indexedAttrs []string, indexAllAttrs bool, logGroupNames []string, skipTimestampValidation bool) ([]string, error) {
+	segments, err := MakeSegmentsFromSpan(span, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+	if err == nil {
+		var documents []string
+
+		for _, v := range segments {
+			document, documentErr := MakeDocumentFromSegment(v)
+			if documentErr != nil {
+				return nil, documentErr
+			}
+
+			documents = append(documents, document)
+		}
+
+		return documents, nil
+	}
+
+	return nil, err
+}
+
+// MakeSegmentsFromSpan creates one or more segments from a span
+func MakeSegmentsFromSpan(span ptrace.Span, resource pcommon.Resource, indexedAttrs []string, indexAllAttrs bool, logGroupNames []string, skipTimestampValidation bool) ([]*awsxray.Segment, error) {
+	var isLocalRoot = false
+
+	if myAwsSpanKind, ok := span.Attributes().Get(awsSpanKind); ok {
+		if localRoot == myAwsSpanKind.Str() {
+			isLocalRoot = true
+		}
+	}
+
+	if !isLocalRoot {
+		segment, err := MakeSegment(span, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+		if err != nil {
+			return nil, err
+		}
+
+		return []*awsxray.Segment{segment}, nil
+	}
+
+	var hasDependency = (span.Kind() != ptrace.SpanKindServer) && (span.Kind() != ptrace.SpanKindInternal)
+	var segments []*awsxray.Segment
+
+	var serviceSegmentID = newSegmentID()
+
+	// If it is a dependency, then we need to create a subsegment
+	if hasDependency {
```

Comment:
Advise moving contents of `hasDependency` into a helper method to make this easier to parse.